src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,22 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> uniqueCharsSet = new TreeSet<>();
+        StringBuilder result = new StringBuilder();
+        BufferedReader reader = new BufferedReader(new FileReader(fileName));
+
+        int symbol = reader.read();
+        while (symbol > 0) {
+            if (Character.isAlphabetic(symbol)) {
+                uniqueCharsSet.add(Character.toLowerCase((char) symbol));
+            }
+            symbol = reader.read();
+        }
+        for (char c : uniqueCharsSet) {
+            result.append(c);
+        }
+
+        return result.length() > 5 ? result.substring(0, 5) : result.toString();
```

Comment:
5 is a magic number here
also, consider checking in a loop if the maximum length is exceeded